### DISCUSSION TITLE
How should `martyrology.ts` be treated?

### DISCUSSION THREAD

**Discussion by TobiTenno** [romcal/romcal#680] [Q&A]
Spinning off a discussion from https://github.com/romcal/romcal/pull/664#discussion_r1792199395.

There is disagreement if `martyrology.ts` should be treated as representing the base/common Roman martyrology that is added to/supplanted by particular calendars OR if it should be treated as a collation/union of all martyrology entries from across all of the particular calendars and including the base Roman Martyrology (which would then be represented purely by the particular rite calendar and not technically be the martyrology, but only the calendar built from that martyrology).

**Comment by TobiTenno** [comment]
My particular opinion is that it should represent the base common Roman Martyrology; such that particular locales/particular calendars define additional entries to the martyrology on their own; rather than a "bas collection" martyrology gathering from across the particular calendars in a way.

**Comment by tukusejssirs** [comment]
What should be in the celebration catalogue?

- All information should be sourced by referencing a source already defined in the sources (source definitions are out of scope of this discussion).
   - There should be a single main source for the whole catalogue item, with optional overrides per property in the catalogue item.
- From the metadata POV: there should be everything necessary to generate celebration names in any format (see #154; cf #676); here are the most relevant stuff (it is not a complete list yet):
   - personal celebrations (single persons):
      - [required] names:
         - this should be an ID a list of names (enum or `const x = {...} as const`), as they should be localisable;
         - it would be nice to include original names, including the name locale;
      - [required] surnames:
         - there should be an option to mark a surname to be localisable, as most surnames are not localisable;
         - moreover in some languages (e.g. Czech or Slovak) we traditionally append `-ová` to female surnames, even foreign ones (e.g. `Jane Smith` → `Jane Smithová`);
         - also keep in mind that in Spanish and Portuguese, people get surnames from both parents, however, there are in different order (in Portuguese, the [first one is the mother’s surname](https://www.researchgate.net/figure/Illustrative-example-of-construction-of-official-surnames-in-Portuguese-speaking-countries_fig2_275582321), [in Spanish, the father’s one](https://en.wikipedia.org/wiki/Spanish_naming_customs));
         - it would be nice to include the original surname, including the surname locale;
      - [required] birthdate (at least a century);
      - [required] death date (at least a century);
      - [required] death date (at least a century);
      - [required] canonisation level:
         - maybe we should rename this to _canonisation stage_ (cf [this](https://www.franciscanmedia.org/ask-a-franciscan/stages-in-the-canonization-process/));
      - [required] sex;
      - [required] list of all common titles that can be used for this person (e.g. `bishop`, `priest`, `martyr`:
         - note that a saint can be inscribed in different calendars under different common titles, therefore they must be chosen in the celebration definitions (in calendars) from this list only;
         - no patronage titles should be defined in the catalogue, as I think they are better to be defined in the celebration definition;
      - [required for bishops] bishopric metadata (at least the diocese name);
      - [required for popes] papal metadata (at least the papal name);
      - [required for religious] religious metadata (at least the religious name);
      - [required for martyrs] martyr metadata (this might not be required by any celebration name format in any locale);
   - collective personal names (more than one person; incl companions):
      - [optional] list of all persons in the collective if they are known;
      - [optional] number of persons (number or `many`; default: `many`; should be defined when the number is known);
      - [required] sex (`male`, `female` or `mixed`);
      - [required] collective name (main one);
      - [optional] alternative collective name;
   - event celebrations:
      - [required] celebration name.

- IMHO the celebration catalogue should be celebration-agnostic, locale-agnostic, ideally also revision-agnostic and rite-agnostic.
   - The only metadata we should use which is a bit related to a rite/revision are the common titles, however, IMHO that is okay. IMHO we should define a list of common titles that can be used in a celebration of a person, as (1) some persons are celebrated under different titles in different calendars; (2) we could validate the titles used in celebration definitions if they use titles from the catalogue item only.

- Which persons should be included in the catalogue? I think the catalogue should contain all persons from servants of God through venerables through blesseds through saints (all four levels/stages of canonisation). IMHO the catalogue should be a separate, buildable and publishable library with some build options to define what data should be included in the output files: (1) everything; (2) only data (person-wise and metadata-wise) used by any calendars we have in romcal; (3) only data used by a selection of calendars we have in romcal (a selection could be one or more). IMHO this would make Etienne happy (he opposes adding persons and metadata not used by romcal), but also me (I want to add as much metadata as possible); moreover it would allow others to limit the data to calendars (and locales, but that is a separate issue for another discussion, possibly) used by them in their software.
   - In the future, I wish to add related texts from the _Roman Martyrology_ for all persons/celebrations inscribed in them, possibble via a separate plugin. Even adding a picture (e.g. encoded in BASE64) of persons/celebrations would be nice, but again the data could be moved to a separate plugin.

**Reply by TobiTenno** [reply]
only the last 2 bullets really have anything to do with this, given that the rest of it is just data fields, and we're covering those data discussions elsewhere, i won't be addressing those points at all.

- titles we already define in an enum, if we want to expand that, that's very doable
- validation is great, but also not really the scope of the discussion prompt

can you say _why_ you believe all of them should be included, even when only used on a specific calendar; or as might happen when adding further rites, those venerated persons may not be recognized in a given martyrology or any such. 

My points center around the Roman Martyrology, because not all the people we will add to romcal would necessarily be in the Roman Martyrology, and I am of the opinion that we should keep the base romcal martyrology to just those that would be in the Roman Martyrology. So, in the cases of persons venerated in, say, Byzantium or Eastern Orthodoxy, we can make it possible to mix in Celebrations from the Roman Martyrology without requiring all of the saints to be on every calendar

**Reply by tukusejssirs** [reply]
First of all, by _all persons/celebrafions_ I mean _all Roman Catholic persons in different stages of canonisation_, at least for now. Yes, saints of other rites would need to be added after we support of their calendars. 

> only the last 2 bullets really have anything to do with this
>
> can you say _why_ you believe all of them should be included, even when only used on a specific calendar; or as might happen when adding further rites, those venerated persons may not be recognized in a given martyrology or any such.

Actually, the second point (partially) answers the question why I suggest to include persons/celebrations outside those added in the _Roman Martyrology_. But I agree I went a bit too far in detail, so that you might missed it: _there should be everything necessary to generate celebration names in any format (see Separation of localisation keys into separate metadata #154; cf Update `MartyrologyDef` #676)_.

The point is, there is much duplication in celebration names: whether it is a common title, canonisation level/stage (or other parts of celebration names), it is written all over again. That’s I suggested to separate celebration names by individual piece of metadata (which Etienne partially implemented in `martyrology.ts`), so that we could generate the celebeation names from metadata. That would provide us the benefit of consistency and reduce the number of potential typos and formatting inconsistency.

On top of that, there is the celebration name formatting. There are several ways to format the celebration name. Each locale and publication has its own preferred format. By separating the metadata (and making the catalogue locale-agnostic), we can format the celebration now however we want or the user wants, for example:

- `${canonisationLevel} ${firstNames} ${surnames}, ${titles}`;
- `${canonisationLevel} ${firstNames} ${surnames}, ${titles} († ${death:year})`;
- `${canonisationLevel} ${firstNames} ${surnames}, ${titles:bishopricSee}`.

Of course, it is harder than having a list of celebration names, but localising and reviewing everything all over again is very tiresome, therefore, one we got it right, it it a pure advantage.

---

I also suggest to add all persons/celebrations added in the _Roman Martyrology_, although Etienne don’t (he wants to add only celebration inscribed to any romcal-supported calendar, which makes sense: he wants to reduce the scope). The reason is that on any weekday/feria, one may choose to celebrate a celebration from the _Roman Martyrology_ inscribe to current day. This is related to the parish calendars (plugins) to generate very unique parish calendar for a particular parish.

---

Finally, I think we should let other apps/libraries (potentially written i different programming languages) to re-use the data. Having a list of all persons venerated in the (Roman) Catholic Vhurch would be awesome. OFC I know many have tried to create such a list (or a website), most of them never succeeded, most never provide sources (which I consider crucial).
 
---

That said, a part of this could be done in a separate plugin, however, unlike calendar or locales, merging the catalogue with some data from the plugin might be too heavy due to its potential size.
